src/server/xtra1.c

```diff
@@ -413,38 +508,206 @@ static void prt_stun(int Ind)
 	player_type *p_ptr = Players[Ind];
 
 	int s = p_ptr->stun;
+	int r;
+
+	if (s > 100)
+	{
+		r = 3;
+	}
+	else if (s > 50)
+	{
+		r = 2;
+	}
+	else if (s)
+	{
+		r = 1;
+	}
+	else
+	{
+		r = 0;
+	}
+
+	send_indication(Ind, IN_STUN, r);
+}
+
+/*
+ * Hack - Display the status line
+ */
+int cv_put_str(cave_view_type* dest, byte attr, cptr str, int col, int max_col)
+{
+	int i;
+	for (i = 0; i < max_col; i++)
+	{
+		dest[i+col].a = attr;
+		dest[i+col].c = str[i];
+	}
+	return 1;
+}
+void c_prt_status_line(player_type *p_ptr, cave_view_type *dest, int len)
+{
+	char buf[32];
+	int col = 0;
+	int i, a;
+	
+	/* Clear */
+	for (i = 0; i < len; i++)
+	{
+		dest[i].a = TERM_WHITE;
+		dest[i].c = ' ';
+	}
+
+	/* Hungry */
+	/* Fainting / Starving */
+	if (p_ptr->food < PY_FOOD_FAINT)
+		cv_put_str(dest, TERM_RED, "Weak  ", COL_HUNGRY, 6);
+	/* Weak */
+	else if (p_ptr->food < PY_FOOD_WEAK)
+		cv_put_str(dest, TERM_ORANGE, "Weak  ", COL_HUNGRY, 6);
+	/* Hungry */
+	else if (p_ptr->food < PY_FOOD_ALERT)
+		cv_put_str(dest, TERM_YELLOW, "Hungry", COL_HUNGRY, 6);
+	/* Normal */
+	else if (p_ptr->food < PY_FOOD_FULL)
+		cv_put_str(dest, TERM_L_GREEN, "      ", COL_HUNGRY, 6);
+	/* Full */
+	else if (p_ptr->food < PY_FOOD_MAX)
+		cv_put_str(dest, TERM_L_GREEN, "Full  ", COL_HUNGRY, 6);
+	/* Gorged */
+	else
+		cv_put_str(dest, TERM_GREEN, "Gorged", COL_HUNGRY, 6);
+
+	/* Blind */
+	if (p_ptr->blind)
+		cv_put_str(dest, TERM_ORANGE, "Blind", COL_BLIND, 5);
+
+	/* Confused */
+	if (p_ptr->confused)
+		cv_put_str(dest, TERM_ORANGE, "Confused", COL_CONFUSED, 8);
+
+	/* Afraid */
+	if (p_ptr->poisoned)
```

Comment:
Does poison make the player afraid, or just certain poison? Is this intentional?